dvc/repo/get_url.py

```diff
@@ -16,6 +16,7 @@ def get_url(url, out=None):
 
     (dep,) = dependency.loads_from(None, [url])
     (out,) = output.loads_from(None, [out], use_cache=False)
-    dep.save()
+    if in_repo:
+        dep.save()
```

Comment:
The unfortunate issue with not calling `save` is that our exceptions in the logic down below are not well unified, so it won't present a nice error when, for example, the source doesn't exist :slightly_frowning_face: I.e. `download` will trigger a method in one of the *Tree classes (in dvc/tree/*), and they all might throw a different backend-specific error, that is pretty ugly :slightly_frowning_face: We are on our way to unifying tree exceptions, but we are not there yet.

Might be missing something, but the only reason we do this `save()` call here is to check if the source exists. If we replace this with `if not dep.exists(): raise DependencyDoesNotExistError(...)`, that should do the trick for now

Also, if I understand correctly, you've introduced this flag to avoid changing the API behavior, right? So like for a backward compatibility? If so, we could totally simply switch to the new behaviour and remove it.

Reply:
You're right. By introducing this flag, we can keep the original logic unaffected.

Reply:
Makes sense. So let's get rid of it and just replace
```
if in_repo:
    dep.save()
```
with
```
if not dep.exists():
    raise dep.DoesNotExistError(dep)
```

Reply:
Changes committed. Seems nothing happened. ^_^